core/src/test/scala/integration/kafka/api/AuthorizerIntegrationTest.scala

```diff
@@ -405,6 +408,34 @@ class AuthorizerIntegrationTest extends KafkaServerTestHarness {
   }
 
   @Test
+  def testPatternSubscriptionWithNoTopicAccess() {
+    addAndVerifyAcls(Set(new Acl(KafkaPrincipal.ANONYMOUS, Allow, Acl.WildCardHost, Write)), topicResource)
+    sendRecords(1, tp)
+    removeAllAcls()
+
+    addAndVerifyAcls(Set(new Acl(KafkaPrincipal.ANONYMOUS, Allow, Acl.WildCardHost, Read)), groupResource)
+    try {
+      this.consumers.head.subscribe(Pattern.compile(topicPattern), new NoOpConsumerRebalanceListener)
+      consumeRecords(this.consumers.head)
```

Comment:
Ah..Thanks for the catch.  with above change, we are subscribing to empty topic list and test case is trying to consume the data. Updated the test case.
Addressed other review comments.